### PR TITLE
A generic class named StorageImpl is implemented, which implements th…

### PR DESCRIPTION
**Pull request by rafik2211**
Storage

A generic class named StorageImpl is implemented, which implements the Storage interface for key-value storage.
It defines a maximum size (MAX_SIZE) for the storage, which is set to 10. It uses two arrays, 'keys' and 'values', to store key-value pairs, both initially created with a size of MAX_SIZE. The 'size' variable keeps track of the number of elements currently in the storage. The put(K key, V value) method allows adding key-value pairs to the storage. It first checks if the key is null and, if so, searches for a null key in the storage. If the key is not null, it looks for an existing key in the storage. If the key is found, the method updates the associated value. If the key is not found, and the storage has room (size < MAX_SIZE), a new key-value pair is added to the arrays. The get(K key) method retrieves a value from the storage based on the provided key. It checks if the key is null and, if so, searches for a null key. If the key is not null, it searches for the key in the storage and returns the associated value. If the key is not found, it returns null. The size() method returns the current size of the storage, indicating how many key-value pairs are stored.

### CODE REVIEW

File: src/main/java/core/basesyntax/Storage.java [hunk 1]
```diff
@@ -1,6 +1,7 @@
 package core.basesyntax;
 
 public interface Storage<K, V> {
+
```
Comment (kozhukhovsky) [inline src/main/java/core/basesyntax/Storage.java:4]:
```suggestion
```

File: src/main/java/core/basesyntax/impl/StorageImpl.java [hunk 2]
```diff
@@ -3,17 +3,57 @@
 import core.basesyntax.Storage;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+    private static final int MAX_SIZE = 10;
+    private Object[] keys = new Object[MAX_SIZE];
+    private Object[] values = new Object[MAX_SIZE];
+    private int size = 0;
```
Comment (kozhukhovsky) [inline src/main/java/core/basesyntax/impl/StorageImpl.java:9]:
```suggestion
    private int size;
```

0 is default value for int

File: src/main/java/core/basesyntax/impl/StorageImpl.java [hunk 3]
```diff
@@ -3,17 +3,57 @@
...
+
     @Override
     public void put(K key, V value) {
+        if (key == null) {
+            for (int i = 0; i < size; i++) {
+                if (keys[i] == null) {
```
Comment (kozhukhovsky) [inline src/main/java/core/basesyntax/impl/StorageImpl.java:15]: You can combine these two checks into one

Comment (kozhukhovsky) [reply src/main/java/core/basesyntax/impl/StorageImpl.java:15]: key == keys[i] || key != null && key.equals(keys[i])

File: src/main/java/core/basesyntax/impl/StorageImpl.java [hunk 4]
```diff
@@ -3,17 +3,57 @@
...
+                    values[i] = value;
+                    return;
+                }
+            }
+        }
+
+        if (size < MAX_SIZE) {
+            keys[size] = key;
+            values[size] = value;
+            size++;
```
Comment (kozhukhovsky) [inline src/main/java/core/basesyntax/impl/StorageImpl.java:32]: This is better to do anyway. But it turns out that the method worked without errors, but the result did not change.
It is necessary that if it is already greater than the maximum value, throw some kind of error.

File: src/main/java/core/basesyntax/impl/StorageImpl.java [hunk 5]
```diff
@@ -3,17 +3,57 @@
...
+        }
     }
 
     @Override
     public V get(K key) {
-        return null;
+        if (key == null) {
+            for (int i = 0; i < size; i++) {
+                if (keys[i] == null) {
```
Comment (kozhukhovsky) [inline src/main/java/core/basesyntax/impl/StorageImpl.java:40]: same, combine

**Review by kozhukhovsky — CHANGES_REQUESTED** [review]

File: src/main/java/core/basesyntax/Storage.java (hunk 1 above)
Comment (Ivan95kos) [reply src/main/java/core/basesyntax/Storage.java:4]: not fixed

File: src/main/java/core/basesyntax/impl/StorageImpl.java [hunk 6]
```diff
@@ -3,17 +3,41 @@
 import core.basesyntax.Storage;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+    private static final int MAX_SIZE = 10;
+    private Object[] keys = new Object[MAX_SIZE];
+    private Object[] values = new Object[MAX_SIZE];
```
Comment (Ivan95kos) [inline src/main/java/core/basesyntax/impl/StorageImpl.java:8]: Use the parameterized types
All non-constant fields should be initialized in the constructor
```suggestion
    private K [] keys;
    private V [] values;
```

File: src/main/java/core/basesyntax/impl/StorageImpl.java (hunk 4 above)
Comment (Ivan95kos) [reply src/main/java/core/basesyntax/impl/StorageImpl.java:23]: not fixed

**Review by Ivan95kos — CHANGES_REQUESTED** [review]
please, fix the previous comments

**Review by kozhukhovsky — APPROVED** [review]

**Pull request closed without merging**